nailgun/entities.py

```diff
@@ -4749,11 +4749,15 @@ def read(self, entity=None, attrs=None, ignore=None, params=None):
         ignore.add('sync_plan')
         result = super(Product, self).read(entity, attrs, ignore, params)
         if 'sync_plan' in attrs:
```

Comment:
Would `if 'sync_plan' in attrs and attrs.get('sync_plan_id')` be the same thing?

Reply:
no they are not? the sync_plan is a dict with attributes  in normal situation I had to Create the SyncPlan object from syn_plan attr but to not break compatibility and as sync_plan_id, used the same logic as it was before.

Reply:
also If we do it like that, we should make many if and elif with the same " 'sync_plan' in attrs"
```python
sync_plan_id = attrs.get('sync_plan_id')
if 'sync_plan' in attrs and sync_plan_id:
     result.sync_plan = SyncPlan(
                    server_config=self._server_config,
                    id=sync_plan_id,
                    organization=result.organization,
                )
elif  'sync_plan' in attrs and not sync_plan_id:
     result.sync_plan = None



```